Marlin/src/HAL/AVR/fast_pwm.cpp

```diff
@@ -181,11 +185,16 @@ void set_pwm_duty(const pin_t pin, const uint16_t v, const uint16_t v_size/*=255
     digitalWrite(pin, !invert);
   else {
     Timer timer = get_pwm_timer(pin);
-    if (timer.isProtected) return;                            // Leave protected timer unchanged
     if (timer.isPWM) {
-      _SET_COMnQ(timer.TCCRnQ, SUM_TERN(HAS_TCCR2, timer.q, timer.q == 2), COM_CLEAR_SET + invert);   // COM20 is on bit 4 of TCCR2, so +1 for q==2
-      const uint16_t top = timer.n == 2 ? TERN(USE_OCR2A_AS_TOP, *timer.OCRnQ[0], 255) : *timer.ICRn;
-      _SET_OCRnQ(timer.OCRnQ, timer.q, uint16_t(uint32_t(v) * top / v_size)); // Scale 8/16-bit v to top value
+      if (timer.n == 0) {
+        TCCR0A |= _BV(COM0B1); // Only allow a TIMER0B select and OCR0B duty update for pin D4 outputs no frequency changes are permited.
+        OCR0B = v;
+      }
+      else {
```

Comment:
Should this instead be … `else if (timer.isProtected) return` to cover pins with `false, true` flags?

Reply:
> Should this instead be … `else if (!timer.isProtected)` to cover pins with `false, true` flags?

Yes, that would be a better assertion.